localMatchApp/headers/homewindow.h

```diff
@@ -2,24 +2,76 @@
 #define HOMEWINDOW_H
 
 #include <QDialog>
+#include "ClassEvent.h"
 
 /*!
  * The class of user interface type is created to
  * be able to develop the main screen
  */
+
 namespace Ui {
 class HomeWindow;
 }
 
-class HomeWindow : public QDialog {
-  Q_OBJECT
+class HomeWindow : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit HomeWindow(QWidget *i_parent = nullptr);
+    ~HomeWindow();
 
- public:
-  explicit HomeWindow ( QWidget *i_parent = nullptr );
-  ~HomeWindow ( );
+/*!
+ * The functions of button actions are declared so
+ * they can be used later
+ */
+private slots:
+    /*!
+     * The function when the user click the button of previous event is declare
+     */
+    void onButtonPreviousClicked();
+    /*!
+     * The function when the user click the button of next event is declare
+     */
+    void onButtonNextClicked();
+    /*!
+     * The function when the user click the button of Subscribe is declare
+     */
+    void onPushButtonSubscribeClicked();
 
- private:
+private:
   Ui::HomeWindow *ui;
```

Comment:
Why you use a raw pointer here?

Reply:
That part the program creates it alone, it is automatic